### ISSUE TITLE
Support for NestJS v10 and @nestjs/graphql v12

### ISSUE THREAD

**Issue by rangelfinal** [MichalLytek/typegraphql-nestjs#51]
Currently, when trying to install this lib a new nest v10 app created via `nest new`, I get resolution issues:

```
code ERESOLVE
ERESOLVE unable to resolve dependency tree

While resolving: backend@0.0.1
Found: @nestjs/common@10.0.3
node_modules/@nestjs/common
  @nestjs/common@"^10.0.3" from backend@0.0.1
  backend
    backend@0.0.1
    node_modules/backend
      workspace backend from the root project

Could not resolve dependency:
peer @nestjs/common@"^9.2.1" from typegraphql-nestjs@0.5.0
node_modules/typegraphql-nestjs
  typegraphql-nestjs@"0.5.0" from backend@0.0.1
  backend
    backend@0.0.1
    node_modules/backend
      workspace backend from the root project

Fix the upstream dependency conflict, or retry
this command with --force, or --legacy-peer-deps
to accept an incorrect (and potentially broken) dependency resolution.
```

After forcing resolution with `npm i --legacy-peer-deps` I had to add `@apollo/subgraph` to my package.json deps (And `graphql-fields` and `graphql-scalars` as well for `typegraphql-prisma`)
That let me to get my app to start and emit GraphQL schema definitions as expected

**Comment by rangelfinal** [comment]
I've tried a local fork to update this package deps with no success
Following the migration guide on https://docs.nestjs.com/graphql/migration-guide, I've tried just replacing `apollo-server-express` with `@apollo/server` and upgrading all deps with `npx npm-upgrade`
But that results in the following error on my app bootstrap:

```
Error: Nest can't resolve dependencies of the GraphQLModule (?, GqlModuleOptions, AbstractGraphQLDriver, GraphQLTypesLoader, GraphQLSchemaHost). Please make sure that the argument HttpAdapterHost at index [0] is available in the GraphQLModule context.

Potential solutions:
- Is GraphQLModule a valid NestJS module?
- If HttpAdapterHost is a provider, is it part of the current GraphQLModule?
- If HttpAdapterHost is exported from a separate @Module, is that module imported within GraphQLModule?
  @Module({
    imports: [ /* the Module containing HttpAdapterHost */ ]
  })
```
Searching on @nestjs/graphql, I've found this maybe related issue: https://github.com/nestjs/graphql/issues/1739
But all @nestjs packages seem to be correctly configured as peerDeps and `npm why @nestjs/common` reveals there's only one version of the common package installed (v10.0.3), so I'm not sure what could be the problem here - maybe --legacy-peer-deps does something very weird that breaks compatibility

**Comment by MichalLytek** [comment]
It's something with npm, peer deps and node_modules. I recommend you install fresh packaged (delete lock file and node_modules), then it updates without any conflicts and run without any issues.

Released as `v0.6.0` 🚀
